core/arch/arm/plat-rockchip/platform_rk3588.c

```diff
@@ -63,3 +81,61 @@ int platform_secure_ddr_region(int rgn, paddr_t st, size_t sz)
 
 	return 0;
 }
+
+TEE_Result hw_get_random_bytes(void *buf, size_t blen)
+{
+	vaddr_t trng_s_base = (vaddr_t)phys_to_virt_io(TRNG_S_BASE,
+						       TRNG_S_SIZE);
+	uint32_t *rand_buf = (uint32_t *)buf;
+	size_t remaining = blen;
+	uint32_t val;
+
+	if (!trng_s_base)
+		panic("TRNG_S base not mapped");
+
+	/* Ensure TRNG is seeded and ready */
+	val = io_read32(trng_s_base + TRNG_S_STAT);
+	if (!(val & (1 << 9))) {
```

Comment:
Please use `BIT32(9)` instead of shifting. By the way, `9` is a magic number, please add and use a define.